index.qmd

```diff
@@ -0,0 +1,65 @@
+---
+title: "Arrow New Feature Showcase: `show_exec_plan()` "
+format: html
+editor: visual
+---
+
+The `arrow` package allows you to take advantage of the power of the [Acero execution engine](https://arrow.apache.org/docs/cpp/streaming_execution.html) for data manipulation and analysis. The code in `arrow` provides bindings to `dplyr` verbs and `tidyverse` functions, so that you can use these interfaces without having to understand the inner workings of Acero. But what if you actually want to know more?
+
+In the latest release of `arrow`, version 9.0.0, the [`show_exec_plan()`](https://arrow.apache.org/docs/r/reference/show_exec_plan.html) function is introduced. This function allows you to see the execution plan generated from your code. For example:
+
+```{r}
+#| echo: false
+#| output: false
+library(arrow)
+library(dplyr)
+```
+
+```{r}
+library(arrow)
+library(dplyr)
+mtcars %>%
+  arrow_table() %>%
+  filter(mpg > 20) %>%
+  mutate(wt_kg = (wt*1000) * 0.453592)  %>%
+  show_exec_plan()
+```
+
+This functionality is similar to that of [`dplyr::explain()`](https://dplyr.tidyverse.org/reference/explain.html) and you'll get the same result whether you call [`arrow::show_exec_plan()`](https://arrow.apache.org/docs/r/reference/show_exec_plan.html), [`dplyr::explain()`](https://dplyr.tidyverse.org/reference/explain.html) or [`dplyr::show_query()`](https://dplyr.tidyverse.org/reference/explain.html) on an `arrow_dplyr_query` object.
```

Comment:
My vision was that `show_exec_plan()` will output the raw form of the plan, while `explain()` and `show_query()` would later evolve into more human readable outputs.

Reply:
Perhaps you could add a sentence mentioning this and linking to the JIRAs for those updates?